src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -235,38 +247,56 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 // After all, the pointer may be lazily initialized outside this initial range.
                 data
             }
-            Err(_) => {
+            _ => {
                 assert_eq!(ptr_size, Size::ZERO); // we did the deref check above, size has to be 0 here
                 // This pointer doesn't come with an AllocId, so there's no
                 // memory to do retagging in.
+                let new_prov = place.ptr().provenance;
                 trace!(
                     "reborrow of size 0: reference {:?} derived from {:?} (pointee {})",
-                    new_tag,
+                    new_prov,
                     place.ptr(),
                     place.layout.ty,
                 );
                 log_creation(this, None)?;
                 // Keep original provenance.
-                return interp_ok(place.ptr().provenance);
+                return interp_ok(new_prov);
             }
         };
+
+        let is_wildcard = matches!(parent_prov, ProvenanceExtra::Wildcard);
+
+        if is_wildcard && ptr_size == Size::ZERO {
+            let new_prov = Provenance::Wildcard;
+            trace!(
+                "reborrow of size 0: reference {:?} derived from {:?} (pointee {})",
```

Comment:
If it doesn't affect your tests, I'd suggest we just early return here for all wildcard reborrows -- let's focus on their "real" accesses for now.